### DISCUSSION TITLE
Auto Bed Leveling issue

### DISCUSSION THREAD

**Discussion by GreatOakIgneous** [Jyers/Marlin#461] [Q&A] [answered]
Hello, I have just installed the E3V2-BLTouch-5x5-v4.2.2.bin, on my Ender 3 V2, but first thing I tested the Leveling tab to auto level the bed it will go to the middle and test the bed and then when it is going to its first position it will go past the bed on the left side and try to check that position. Idk if this would be considered a bug but I have attached two videos (since I was 4mb to large for one) to show what is happening. I have tried other bin files like the High speed 5x5 and the 3x3, and other versions but they all do the same thing when I try leveling the bed. 

https://user-images.githubusercontent.com/82343975/114323843-57220380-9ae4-11eb-8537-c6ee02e8ade5.mov

https://user-images.githubusercontent.com/82343975/114323850-5c7f4e00-9ae4-11eb-8b3c-417733a13f4e.mov

**Comment by aryajuanda** [comment, marked as answer]
you need to set probe to nozzle offset so your probe stay in the bed,
for official bltouch bracket its around -44 for X and -7 for Y, you can set it on advanced menu.. or you can compile with your on probe to nozzle offset.

**Reply by RPhilbo** [reply]
Thank you, that helped me as well. 👍 :-)

**Reply by kazlyn-1** [reply]
> 
> 
> Thank you, that helped me as well. 👍 :-)

And me, thanks a lot 👍

**Reply by MehItsOk** [reply]
Thanks,

**Reply by ssshake** [reply]
I have this issue as well. What's strange to me is that the official ender firmware works fine with the default of -10, -10. I'm just curious why there's a difference.

**Reply by ssshake** [reply]
This might be a good detail for the wiki. I'm brand new to the ender 3 v2. I have a stock printer + a bltouch. This was a bit of a head scratcher especially because it wasn't an issue with the stock firmware. I did look at the docs, this could have saved me some time and also spared people like us creating these discussions.

**Reply by aryajuanda** [reply]
Maybe @Jyers can made this value as default for bltouch series firmware, because its has be a common issue for new person who try this firmware, because there is no information about this in the wiki.. i personaly set the value in the config and build it manually with some adjustment config... but not everyone build their own firmware..

**Reply by ssshake** [reply]
Well I have a few thoughts on this. 

1) The issue from my end seems to have spawned from different behavior between the official firmware and this firmware WITH the same values entered. 

The take away is from the perspective of the novice, going from the stock firmware to this, it can throw them for a loop.

2) Why have the default set to -10,-10? Why not have it just set to 0,0? Having a value set makes it feel like it's supposed to be there for a reason. Being a novice can work against better judgement sometimes when you don't think you know enough to challenge what you see. 

3) As for the wiki, I was thinking something as simple as "BTW your x/y offset for bltouch is going to be wrong so you should measure it and set it".

1.3.1 just launched and had nothing to do with this. Yet here we are, 4 of us in just two weeks have gotten to the point where they felt like they needed to actually post something, specifically about this firmware version. 

I assume we posted here because 1.3.1 is new, and that's what we loaded, and then we saw this original post. I wonder how many times in the future this pattern will repeat. 

So maybe just for sanity on your end, document something so that people won't post threads like this in error. 

Ultimately this thread did help me because the logical path I took to get here was probably the same as the other people who ended up here. 

That path went something like this:

- I Buy an ender 3 v2.
- I install a BLtouch.
- Life it pretty good but the stock firmware is lacking advanced on-screen settings.
- I learn about this custom firmware and get excited.
- I load it up, I'm very happy with the available settings.
- But when I do a bed level suddenly things are going weird. 
- I start searching the firmware's issue tracker and discussion page.
- Here we are.

**Comment by smooreace** [comment]
Its not common at all. There are infinite combinations for the required
measurements

**Reply by aryajuanda** [reply]
What i mean a common issue is there is a lot people dont know about this probe and nozzle things.. that they have to set before they can use ABL.. the combination its so many.., stock hotend with stoct bltouch mount (metal plate) from creatily give that combination..

**Comment by smooreace** [comment]
If people don't follow instructions then they deserve to get nothing but
failure. Trying to define a "default" set of arbitrary offsets would
obviously create more issues than it solves. First, it would assume that
every BLTouch installed is mounted to a given mount, on a given carriage
with a predefined cooler. The people that actually fall into that
category should already know to do what the instructions tell them to do
and would do nothing but lead to confusion when they attempt to follow
those instructions. Then the tens of thousands of other people would be
screwed by some random garbage data that s inserted where there should be
zeros.

Or are you suggesting that any company manufacturing a variant of the BLT
should provide 2 sets of instructions, one for people to do their own work,
and one for fools not smart enough to follow them?

**Reply by ssshake** [reply]
The firmware target in question is quite specific. It's for an Ender 3 V2 with BL Touch, as called out by name in the list of releases: https://github.com/Jyers/Marlin/releases/tag/v1.3.1

Creality sells an official BL touch kit. The clones ship with brackets for the Ender 3 V2. 

It's pretty reasonable to assume these default settings should match the default settings of the official firmware, since people are migrating from that firmware to this. This would reduce some friction in the migration, simply because they are expecting the values to be the same. After all this is a custom firmware specific to the exact printer they have. It isn't generic.

If the settings were the same as the stock firmware, any variation / customization by the end user would be subject to the same changes across both the official and unofficial firmware.

**Comment by ssshake** [comment]
Just wanted to add some data points here. This is the second facebook group post in the past week where I've helped someone fix the BLtouch offset problem after migrating to jyers.

I simply tell them to set their offset to -44, -7. 

These questions and the time spent by the novice troubleshooting them could be eliminated if the firmware had a better default than -10, -10 or if the WIKI explained this. 

People expect when migrating from stock to jyers that something like the bltouch probe defaults will be the same. When it's different they assume they did something wrong. As this person is calling themselves dumb, but they are not. 

![image](https://user-images.githubusercontent.com/1675868/116253858-d9decb80-a73e-11eb-8441-b7948eda5490.png)

**Reply by Jyers** [reply]
The reasoning for setting this to  -10, -10 by default was to encourage people to measure or find their own offset (as many don't and will therefore have a slightly offset mesh) but I suppose the firmware has outgrown this method. I'll change the default.

**Comment by KiLLie2k** [comment]
How would one go about finding the exact offset of the bltouch? Calipers??? I love knowing my calibrations are precise :)  I just used the default from the wiki here, it threw me off for a bit when I first installed the firmware as well lol.  But half the fun of this printer at least for me is learning and tinkering with it!

**Reply by Jyers** [reply]
The best method I've found is to put some tape on the bed and mark the point where the nozzle is. Then use the move menu to move the print head until the probe is over this point. By recording the coordinated before and after moving you can find the offset pretty accurately

**Reply by KiLLie2k** [reply]
i'll give it a go, I was thinking maybe bring the nozzle right to the edge of the bed and then move until the bltouch lines up :) tape is safer lol. Thanks for all your hard work :)
